### ISSUE TITLE
Route/Page Transition

### ISSUE THREAD

**Issue by pxwee5** [vuejs/vuepress#1884]
## Feature request
Is it in the roadmap to add page transitions to VuePress?

#### What problem does this feature solve?
Beautify the transition between pages

**Comment by kefranabg** [comment]
What kind of transition are you thinking about?

**Comment by gevera** [comment]
Fade in and out

**Comment by CodesOfRa** [comment]
I can work on this :)

**Comment by sdaitzman** [comment]
@CodesOfRa are you still planning to work on this? I'm considering writing a plugin from scratch, but don't want to duplicate your work, especially if there's any work I could do to help.

The simplest implementation I've come up with so far would just involve adding a Vue <Transition> component as a parent of the <GlobalLayout>, which (if I understand correctly) should handle timing and expose the right CSS classes using the default Vue transition system.

**Comment by sdaitzman** [comment]
A much-refined proof-of-concept version of the idea I describe, written by @jdtzmn as a patch rather than a plugin, is here: https://github.com/jdtzmn/vuepress

This implementation would also meet #2098, I think (@Axieum confirm?). It allows for any custom root combination of Vue components to replace the default router setup.

Tomorrow, I'll try to get an example live. If this is a useful feature, I can document/test/refactor @jdtzmn's code as needed.

**Comment by Axieum** [comment]
Yes, this appears to solve #2098, however will need modifying to remove a couple set in stone elements; see comment on commit jdtzmn/vuepress@aa638ee8411cacb28f51f1ef42acfbc12ae23a78#r36658709

**Comment by sdaitzman** [comment]
@Axieum can you describe your use case for those additional modifications?

**Comment by Axieum** [comment]
I just implemented `Content` level transitions very easily. I'm assuming this issue is related to being able to transition upper level elements closer to the `#app`?

**Comment by sdaitzman** [comment]
The issue is that, to implement smooth transitions between different routes (and, therefore, layouts) requires including a direct parent of the router-view Vue element and specifying transitions in the same Vue file.

**Comment by CodesOfRa** [comment]
Hi 👋 @sdaitzman  I am still planning to work on this, but I am a bit busy at the moment so if you have the time feel free to take it over

**Comment by sdaitzman** [comment]
@CodesOfRa thanks for letting me know! Do you have any work-in-progress somewhere that I could start from?
